plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -292,10 +297,21 @@ func (s *serviceAccount) limitSecretReferences(serviceAccount *api.ServiceAccoun
 			continue
 		}
 		secretName := source.Secret.SecretName
-		if !referencedSecrets.Has(secretName) {
+		if !mountableSecrets.Has(secretName) {
```

Comment:
> Hrm, you mean I can't add a secret manually and then use it? I have to add it, add a ref to the service account, and then use it?

It depends on whether you're set the service account admission controller to enforce the restriction.  Used in this manner, service accounts can help control access to secrets in a namespace.  I've extended it to apply similar rules to `ImagePullSecrets`.